lib/matplotlib/colorizer.py

```diff
@@ -701,3 +763,163 @@ def _auto_norm_from_scale(scale_cls):
         norm = colors.make_norm_from_scale(scale_cls)(
             colors.Normalize)()
     return type(norm)
+
+
+def _ensure_norm(norm, n_components=1):
+    if n_components == 1:
+        _api.check_isinstance((colors.Norm, str, None), norm=norm)
+        if norm is None:
+            norm = colors.Normalize()
+        elif isinstance(norm, str):
+            scale_cls = _api.check_getitem(scale._scale_mapping, norm=norm)
+            return _auto_norm_from_scale(scale_cls)()
+        return norm
+    else:  # n_components > 1
+        if not np.iterable(norm):
+            _api.check_isinstance((colors.MultiNorm, None, tuple), norm=norm)
+        if norm is None:
+            norm = colors.MultiNorm(['linear']*n_components)
+        else:  # iterable, i.e. multiple strings or Normalize objects
+            norm = colors.MultiNorm(norm)
+        if isinstance(norm, colors.MultiNorm) and norm.n_components == n_components:
+            return norm
+        raise ValueError(
+                "Invalid norm for multivariate colormap with "
+                f"{n_components} inputs"
+            )
```

Comment:
Should fit together.
```suggestion
            f"Invalid norm for multivariate colormap with {n_components} inputs")
```